File: Tawaf/app/src/main/java/me/mxcsyounes/tawaf/models/Line2D.java
package me.mxcsyounes.tawaf.models;

public class Line2D {

    private float a, b, c;


    public Line2D(float a, float b, float c) {
        this.a = a;
        this.b = b;
        this.c = c;
    }

    public Line2D(Point2D pointOne, Point2D pointTwo) {
        this.a = pointTwo.getY() - pointOne.getY();
        this.b = pointOne.getX() - pointTwo.getX();
        this.c = -(a * pointOne.getX() + b * pointOne.getY());
    }

    public float getC() {
        return c;
    }

    public float getB() {
        return b;
    }

    public float getA() {
        return a;
    }

    public double distanceToPoint(Point2D point) {
        double nominator = this.getA() * point.getX() + this.getB() * point.getY() + this.getC();
        double denominator = Math.sqrt(this.getA() * this.getA() + this.getB() * this.getB());
        return nominator / denominator;
    }

    public double getCoefficientOfGuid() {
        return -this.a / this.b;
    }

    @Override
    public String toString() {

        return String.valueOf(this.getA()) + "x " + String.valueOf(this.getA()) + "y " + String.valueOf(this.getC());
    }
}

File: Tawaf/app/src/main/java/me/mxcsyounes/tawaf/models/Line2D.kt
package me.mxcsyounes.tawaf.models;

class Line2D {

    val a: Float
    val b: Float
    val c: Float

    constructor(a: Float, b: Float, c: Float) {
        this.a = a
        this.b = b
        this.c = c
    }

    constructor(pointOne: Point2D, pointTwo: Point2D) {
        this.a = pointTwo.y - pointOne.y;
        this.b = pointOne.x - pointTwo.x
        this.c = -(a * pointOne.x + b * pointOne.y)
    }

    fun distanceToPoint(point: Point2D): Double {
        val nominator = a * point.x + b * point.y + c
        val denominator = Math.sqrt((a * a + b * b).toDouble());
        return nominator / denominator
    }

    fun getCoefficientOfGuid() = a / b
}
